cajita/unit_test/tstSparseIndexSpace.hpp

```diff
@@ -0,0 +1,681 @@
+    constexpr int size_per_dim = size_tile_per_dim * 4;
+    constexpr int total_size = size_per_dim * size_per_dim * size_per_dim;
+
+    std::array<int, dim_n> size( { size_per_dim, size_per_dim, size_per_dim } );
+    int capacity = size_per_dim * size_per_dim;
+    float rehash_factor = 1.5;
+    SparseIndexSpace<TEST_EXECSPACE> sis( size, capacity, rehash_factor );
+
+    constexpr int insert_cell_num = 100;
+    Kokkos::View<int*, TEST_DEVICE> qid_res( "query_id", insert_cell_num );
+    Kokkos::View<int*, Kokkos::HostSpace> host_cell_1did( "cell_ids_1d",
+                                                          insert_cell_num );
+    std::map<int, int> cell_register;
+    for ( int i = 0; i < insert_cell_num; ++i )
+    {
+        host_cell_1did( i ) = ( std::rand() % total_size );
+        int tile_k = ( host_cell_1did( i ) % size_per_dim ) >> 2;
+        int tile_j =
+            ( ( host_cell_1did( i ) / size_per_dim ) % size_per_dim ) >> 2;
+        int tile_i = ( ( host_cell_1did( i ) / size_per_dim / size_per_dim ) %
+                       size_per_dim ) >>
+                     2;
+        cell_register[tile_i * size_tile_per_dim * size_tile_per_dim +
+                      tile_j * size_tile_per_dim + tile_k] = 1;
+    }
+    int valid_cell_num = cell_register.size();
+    Kokkos::View<int*, TEST_DEVICE> dev_cell_1did( "cell_ids_1d_dev",
+                                                   insert_cell_num );
+    Kokkos::deep_copy( dev_cell_1did, host_cell_1did );
+
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<TEST_EXECSPACE>( 0, insert_cell_num ),
+        KOKKOS_LAMBDA( int cell_idx ) {
+            int id = dev_cell_1did( cell_idx );
+            int k = id % size_per_dim;
+            int j = ( id / size_per_dim ) % size_per_dim;
+            int i = ( id / size_per_dim / size_per_dim ) % size_per_dim;
+            sis.insertCell( i, j, k );
+        } );
+
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<TEST_EXECSPACE>( 0, insert_cell_num ),
+        KOKKOS_LAMBDA( int cell_idx ) {
+            int id = dev_cell_1did( cell_idx );
+            int k = id % size_per_dim;
+            int j = ( id / size_per_dim ) % size_per_dim;
+            int i = ( id / size_per_dim / size_per_dim ) % size_per_dim;
+            auto qid = sis.queryTile( i, j, k );
+            qid_res( cell_idx ) = qid;
+        } );
+
+    bool test[size_per_dim * size_per_dim * size_per_dim];
+
+    auto qid_mirror =
+        Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), qid_res );
+
+    for ( int i = 0; i < total_size; i++ )
+    {
+        test[i] = false;
+    }
+
+    for ( int i = 0; i < insert_cell_num; ++i )
+    {
+        test[qid_mirror( i )] = true;
+    }
+
+    int idx = 0;
+    while ( idx < valid_cell_num )
+    {
+        EXPECT_EQ( test[idx], true );
+        ++idx;
+    }
+    while ( idx < total_size )
+    {
+        EXPECT_EQ( test[idx], false );
+        ++idx;
+    }
+
+    sis.clearIndexSpace();
```

Comment:
Should check `size()` and `capacity()` after this function is called to ensure correctness. `size()` should be zero and `capacity()` should be unchanged.

Reply:
Added